packages/discovery/test/discovery.spec.js

```diff
@@ -0,0 +1,41 @@
+import * as assert from 'assert'
+import * as discovery from '../src/index.js'
+import { invoke } from '@ucanto/core'
+import * as Server from '@ucanto/server'
+import * as CAR from '@ucanto/transport/car'
+import * as principal from '@ucanto/principal'
+
+describe('discovery', () => {
+  testDiscovery(discovery, async (name, test) => it(name, test))
+})
+
+/**
+ * test discovery module
+ * 
+ * @param {typeof discovery} module - discovery module to test
+ * @param {import("./test-types").TestAdder} test - add a named test
+ */
+function testDiscovery(module, test) {
+  test('is an object', async () => {
+    assert.equal(typeof module, 'object')
+  })
+  test('can discovery/assert/location', async () => {
+    const id = await principal.ed25519.generate()
+    const service = discovery.service.create()
+    const server = Server.create({
+      codec: CAR.inbound,
+      id,
+      service,
+    })
+    const claimResult = await server.run(invoke({
```

Comment:
this works at runtime but not tsc time @Gozala 

I modeled it after this test https://github.com/web3-storage/ucanto/pull/274/files/50f00605e89bd57656aaee6de26316e2d385732c#diff-cdde9fc2d29ed4b39d6ac58052c126c2c661ea3a244c844a0d75ee54394aff98R422

Reply:
It's unclear to me what could be the cause here, but I would suggest:

1. Adding a test to ucanto directly that basically does this & if fails just try to simplify it as much as possible. If it does not fail there it could be a TSC version which is 5.x while ucanto is still on 4.9
2. I made comment on the line where you define a provider for the service in regards to some inference considerations. It may be a cause here, but worth trying to eliminate.
3. The `const` annotation below could also be problem. Ucanto functions were written so it they won't be unnecessary, yet perhaps presence does affect it somehow. Worth eliminating this possibility.
4. Call to invoke here is inlined inside a call to `.run` which in TS affects how inference will work and here it could mean that TS considers `service` type to infer the result of `invoke` in a way that causes problems when it later tries to unify types. Might be worth trying not to inline things and trying it that way.

If it's none of the above, I have no idea why would this not work and would recommend going with in-process serve / client setup [like in many other tests](https://github.com/web3-storage/ucanto/blob/b330244d1505141afb5c53f299b9e923e716e495/packages/server/test/handler.spec.js#L109-L149) unlike `.run` code path it had been around for long time and most (ucanto) tests use that setup. That is to say `.run` is new and perhaps has problems.